packages/lib/src/__tests__/format.test.ts

```diff
@@ -0,0 +1,224 @@
+import { formatAndWriteItem, formatAndWriteList } from '../format'
+import { IOFormat } from '../io-util'
+import * as output from '../output'
+import * as outputBuilder from '../output-builder'
+import { TableFieldDefinition, TableGenerator } from '../table-generator'
+import { buildMockCommand } from './test-lib/mock-command'
+import { SimpleType } from './test-lib/simple-type'
+
+
+describe('format', () => {
+	const item = { 'str': 'string', num: 5 }
+	const list = [item]
+	const baseCommand = {
+		...buildMockCommand(),
+		flags: {
+			output: 'output.yaml',
+		},
+	}
+
+	const outputFormatter: jest.Mock<string, [data: unknown]> = jest.fn()
+	const buildOutputFormatterSpy = jest.spyOn(outputBuilder, 'buildOutputFormatter')
+	const writeOutputSpy = jest.spyOn(output, 'writeOutput')
+
+	beforeEach(() => {
+		outputFormatter.mockReturnValue('output')
+		buildOutputFormatterSpy.mockReturnValue(outputFormatter)
+		writeOutputSpy.mockResolvedValue()
+	})
+
+	afterEach(() => {
+		jest.clearAllMocks()
+	})
+
+	describe('formatAndWriteItem', () => {
+		let itemTableFormatterSpy: jest.SpyInstance<output.OutputFormatter<unknown>, [tableGenerator: TableGenerator, fieldDefinitions: TableFieldDefinition<unknown>[]]>
+
+		beforeEach(() => {
+			itemTableFormatterSpy = jest.spyOn(output, 'itemTableFormatter')
```

Comment:
We could also move this out of beforeEach to clear up the long type above.

Reply:
Sorry, I thought I had got them all. Fixed.